internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -282,6 +345,51 @@ func handleParseError(typ string, parseErr error) *metrics.WebhookEventState {
 	return state
 }
 
+func (_ *Server) parseGithubAppEventForProcessing(
+	rawWHPayload []byte,
+	msg *message.Message,
+) error {
+	var event github.InstallationEvent
+
+	if msg.Metadata.Get(events.GithubWebhookEventTypeKey) != "installation" {
+		return newErrNotHandled("github app event %s not handled", msg.Metadata.Get(events.GithubWebhookEventTypeKey))
+	}
+
+	err := json.Unmarshal(rawWHPayload, &event)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling payload: %w", err)
+	}
+
+	action := event.GetAction()
+	if action == "" {
+		return fmt.Errorf("action is empty")
+	}
+
+	if action != "deleted" {
+		return newErrNotHandled("event %s with action %s not handled",
```

Comment:
The error handling / control flow here sets my teeth on edge, but I don't have a better alternative yet.

Reply:
To be specific: it seems like `errNotHandled` is a success-error, which then gets consumed and transformed in some way by `handleParseError`, and the result of that interacts with GitHub's webhook retry-on-error logic.